test/CodeGen/AVR/sign-extension.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc -march=avr < %s | FileCheck %s
+
+define i64 @sign_extended(i1) {
+; CHECK-LABEL: entry-block
+entry-block:
+  %1 = sext i1 %0 to i8
+  %2 = sext i8 %1 to i16
```

Comment:
There's an existing clever solution for sign extending 8-bit to 16-bit:

```asm
        mov     r25, r24
        lsl     r25
        sbc     r25, r25
```

It took me a while to understand it, but it really solves it succinctly, time and space-wise. 

*Conceptually*, we could build on this by copying the upper byte to successively higher positions. The problem is that we still generate a bunch of left-shifts. Presumably this is because of the whole 8-bit 16-bit weirdness that is going on.

I'm now investigating if I can rework the `SEXT` pseudo-op to make it more usable for larger sizes.